### DISCUSSION TITLE
Max buffer size doesn't work

### DISCUSSION THREAD

**Discussion by chenhuan-gao** [miguelgrinberg/Flask-SocketIO#1998] [Q&A] [answered]
**I set app and socketio like this:**
app = Flask(__name__)
app.config['SECRET_KEY'] = 'secret!'
app.config['MAX_CONTENT_LENGTH'] = 1024 * 1024 * 1024
socketio = SocketIO(engineio_logger=True, logger=True, max_http_buffer_size=1024 * 1024 * 1024)
socketio.init_app(app)

and when it still not able to recieve a 8mb file, while a 700kb file can be recieved successfully.

**here is the log when I try to recieve the 8mb file:**
Bv_mju8y7_szCRN9AAAA: Received packet MESSAGE data 51-["upload learner recording",{"file":{"_placeholder":true,"num":0},"name":"8m_test"}]
client disconnected.
127.0.0.1 - - [04/Jul/2023 18:24:31] "GET /socket.io/?EIO=4&transport=websocket&sid=Bv_mju8y7_szCRN9AAAA HTTP/1.1" 200 0 224.093795
client connected.
xeFLV6HYnx9xhgpRAAAC: Sending packet OPEN data {'sid': 'xeFLV6HYnx9xhgpRAAAC', 'upgrades': ['websocket'], 'pingTimeout': 20000, 'pingInterval': 25000}
127.0.0.1 - - [04/Jul/2023 18:24:33] "GET /socket.io/?EIO=4&transport=polling&t=OaWRhDI HTTP/1.1" 200 278 0.001002
xeFLV6HYnx9xhgpRAAAC: Received packet MESSAGE data 0
xeFLV6HYnx9xhgpRAAAC: Sending packet MESSAGE data 0{"sid":"xBWiQxA3uwSbDduWAAAD"}

**and here is log when I received the 700kb file:** 
xeFLV6HYnx9xhgpRAAAC: Received packet MESSAGE data 51-["upload learner recording",{"file":{"_placeholder":true,"num":0},"name":"test_recording"}]
xeFLV6HYnx9xhgpRAAAC: Received packet MESSAGE data <binary>
received event "upload learner recording" from xBWiQxA3uwSbDduWAAAD [/]
emitting event "update recording" to xBWiQxA3uwSbDduWAAAD [/]
xeFLV6HYnx9xhgpRAAAC: Sending packet MESSAGE data 2["update recording",{"image":"/9j/4AAQSkZJRgABAQEAZABkAAD/2wBDAAgGBgcGBQgHBwcJCQgKDBQNDAsLDBkSEw8UHRofHh0aHBwgJC4nICIsIxwcKDcpLDAxNDQ0Hyc5PTgyPC4zNDL/2wBDAQgJCQwLDBgNDRgyIRwhMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjL/wAARCABkA+gDASIAAhEBAxEB/8QAHwAAAQUBAQEBAQEAAAAAAAAAAAECAwQFBgcICQoL/8QAtRAAAgEDAwIEAwUFBAQAAAF9AQIDAAQRBRIhMUEGE1FhByJxFDKBkaEII0KxwRVS0fAkM2JyggkKFhcYGRolJicoKSo0NTY3ODk6Q0RFRkdISUpTVFVWV1hZWmNkZWZnaGlqc3R1dnd4eXqDhIWGh4iJipKTlJWWl5iZmqKjpKWmp6ipqrKztLW2t7i5usLDxMXGx8jJytLT1NXW19jZ2uHi4+Tl5ufo6erx8vP09fb3+Pn6/8QAHwEAAwEBAQEBAQEBAQAAAAAAAAECAwQFBgcICQoL/8QAtREAAgECBAQDBAcFBAQAAQJ3AAECAxEEBSExBhJBUQdhcRMiMoEIFEKRobHBCSMzUvAVYnLRChYkNOEl8RcYGRomJygpKjU2Nzg5OkNERUZHSElKU1RVVldYWVpjZGVmZ2hpanN0dXZ3eHl6goOEhYaHiImKkpOUlZaXmJmaoqOkpaanqKmqsrO0tba3uLm6wsPExcbHyMnK0tPU1dbX2Nna4uPk5ebn6Onq8vP09fb3+Pn6/9oADAMBAAIRAxEAPwD3+iiigAooooAKKKKACiiigAooooAKKKKACiiigAooooAKKKKACiiigAooooAKKKKACiiigAooooAKKKKACiiigAooooAKKKKACiiigAooooAKKKKACiiigAooooAKKKKACis/V9bsNDtfPvpxGDwiDl3Poo71z4ttc8W/NeGXSNIbpbocTzj/AGj/AAj2/wD11rGm2uZ6L+vvMp1UnyrV9v8APsXtR8YWVtdGx0+GXVL/AP54Wo3Bf95ug/XFVxb+MtT+aW9stHjPSOGPz5B7Enj8q3tN0qx0i1FvYW0cEY6hRy3uT1J+tXKftIx+BfN6/wDAJ9nKXxv5LT/g/kct/wAIxrPX/hLb7d6+UuPyo/s3xjac2+u2d6B0W7tfL/VOa6mij20utvuQ/YR6XXzZy39t+KLPi98NrcKOsllcA/8Ajp5o/wCE6souL3TdWsz3860OP0zXU0Uc8HvH7m/+CHs5raX3pP8AyOXHxD8MH72osh9Gt5P/AImlPxD8Lj/mJk/9u8n/AMTXTFVbqAfqKAqjoAPwovS7P7/+AHLV/mX3f8E5gfETwvnnUWH1t5P/AImpV8e+GH6aqn4xuP5rXRkAjBGaia1t3+/BE31QGi9Ls/v/AOAFqvdfc/8AMxG8ceGlGTq0P4Bj/Sq7eP8AQ3JWzN3fP/dtrZiT+YFb40+yU5WztwfURL/hVhVCjCgAegovS7P7/wDgBaq+q+7/AIJy39t+JtR403w+LRD0m1GTbj/gA5o/4RrWdQ51jxHc7T1gsFEKj23dSPqK6qij2tvhSX9eYvY3+Nt/h+Rzlv4E8OQHc2niaQ9XnkZyfrk4q6vhfQFGBo1h+Nup/pWtRUurUe8n95ao01tFfcY7eFfD7ddGsfwhUVC/gvw4/XSLf8AR/I1vUUe1mvtP7w9lTf2V9yObbwD4YbrpSfhLIP8A2aoz8PPDGcrp7L9LiT/4quooqvb1f5n95PsKX8q+45b/AIQDRF/1JvYT6x3LD+dH/CHTQc2XiTWYfRZJhIo/Aiupoo9vU6sPq9Pov6+85b+zPGFp/wAe+vWd6B0W7tdn6pzR/bHiqy/4/fDsV0g6yWVwP/QW5NdTRR7W/wAUU/w/IPY2+GTXzv8AmcunjvSo3EeowX2myHjF3bsoz9Rmt2y1Sw1Jd1lewXA7+XIGI+o7VZeNJUKSIrqeqsMg1hXvgvQL1vMNglvMORJbExMD6/Lx+lF6T7r8f8hWqx6p/h/mjforlP7B8RaXzpGvNcxjpb6ku8H/AIGOaP8AhLL3TPl8QaJcWqDrdW376L6nHK/rR7Fv4Hf8/uYe2S+NNfl96Oroqjp2sadq0fmWF5DcDGSEb5h9R1H41erJpp2ZsmmroKKKKQwooooAKKKKACiiigAooooAKKKKACiiigAooooAKKKKACiiigAooooAKKKKACiiigAooooAKKKKACiiigAooooAKKKKACiiigAooooAKKKKACiiigAooooAKKKKACiiigAooooAKKKKACiiigAooooAKKKKACiiigAooooAKKKKACiiigAooooAKKKKACiiigAooooAKKKKACiiigAooooAKKKKACiiigAooooAKKKKACisvVPEekaMD9uvoo37RA7nP/ARzWR/b+v6vxomim3hPS71E7B9Qg5P1rWNKTV9l56GUqsIu277LU6iWWOCJpZpEjjUZZ3OAB7k1y9x4qudVmez8L2v2pwdr3soKwRfj/Efp+tOj8HNfSrP4i1KfU5AciD/AFcCn/dHWulhgitoVhgiSKJBhURQoA9gKr93D+8/w/4P4E/vJ/3V+P8AkvxMPSfCsFndf2jqMz6jqh5NxMOE9kXooroKKKylOUneRpCEYK0QoooqSwooooAKKKKACiiigAooooAKKKKACiiigAooooAKKKKACiiigAooooAKKKKACiiigAooooAwdR8H6LqMnnfZvstyDkXFq3lOD68cE/UVR+yeLNE5tLuHWrVf+WVz+7mA9A/Q/U11lFaqtK1nqvMxdGN7rR+X9W/A5yy8aabNcC01BZdLvP8AnjeLsB+jdCPyrogQwBBBB6EVXvtPs9StzBe20U8R/hkXOPp6VzbeGdT0QmXw1qBEQ5NheEvEfZT1X/PNVanPbR/h9/8AmK9SG/vLy3+7/I62iuc07xdBLdjT9Wt30vUe0U5+ST/cfof89a6Os5QlB2kaQnGavFhRRRUFhRRRQAUUUUAFFFFABRRRQAUUUUAFFFFABRRRQAUUUUAFFFFABRRRQAUUUUAFFFFABRRRQAUUUUAFFFFABRRRQAUUUUAFFFFABRRRQAUUUUAFFFFABRRRQAUUUUAFFFFABRRRQAUUUUAFFFFABRRRQAUUUUAFFFFABRRRQAUUUUAFFFFABRRRQAUUUUAFFFFABRRRQAUUUUAFFFRzTw20ZknljiQdWdgo/M0ASUVzt3440C2k8qO8N3MekdohlJ+hHH61X/4SDxDqHGl+G5IUPSbUJBHj6oOf1rZUZ7tW9dDF14bJ39NfyOqqlf6vp2lx7769gtx1AdwCfoOp/CsL+wfEeo/8hTxCbeM9YdOj2f8Aj55q5YeDdC09/NWyWefOTNckysT688Z+go5acfilf0/zYuepL4Y29f8AJFM+MZdQOzw/o93qGeBO6+VD/wB9N1+nFJ/YniPV+dX1oWcJ622mjbx7uef511YAAAAwB2oo9qo/Arfi/wCvkHsnL45X/Bf18zI0vwxo+jkPaWUfndTNJ88hPruPI/Cteiis5ScneTuaxjGKtFWCiiipKCiiigAooooAKKKKACiiigAooooAKKKKACiiigAooooAKKKKACiiigAooooAKKKKACiiigAooooAKKKKACiiigAooooAp6lpVjrFo1tf26TRHoGHKn1B6g/SuZJ1jwbyTNquhr1zzPbD/wBmUf5xXZUVrCo4rleq7f1sZTpqT5lo+/8AW5WsNQtNUs0u7KdJoH6Mp/Q+h9qs1yt/4cutNvH1Xwy6wXDcz2TcQ3H4fwt7/wAua0dD8R2utK8Wxra/h4ntJeHQ/wBR705U1bmhqvxXr/mKNR35Z6P8H6f5GzRRRWJsFFFFABRRRQAUUUUAFFFFABRRRQAUUUUAFFFFABRRRQAUUUUAFFFFABRRRQAUUUUAFFFFABRRRQAUUUUAFFFFABRRRQAUUVXmvrS2/wBfdQRf9dJAv86aTewm0tyxRWTJ4n0GI4fWLHPoJ1P8jRVezn2f3Ee1h/MvvRrUUUVBoFFFFABRRRQAUUUUAFFFFABRRRQAUVFLdW8H+unij/33A/nVCbxJocH+s1exU+nnqT+WapRk9kS5RW7NSiuel8c+GovvatEf9xWb+Qqv/wALB0Bv9TLcz/8AXO2c/wAxVqhUf2WZuvSX2l951NFct/wnVk3+q0vWJf8Acsyf60f8Jm7f6vwzr7fWzx/Wn7Cp2/L/ADD29Pv+f+R1NFct/wAJbfN9zwrq5/3owtH/AAlGrn7vhO//ABkUUewn/TX+Ye3h/Sf+R1NFct/wkmvHp4RuvxuUH9KP+Ej1/wD6FG5/8Ck/wo9jPy+9f5h7eHn9z/yOporlv+Ek10dfCN1+Fyh/pR/wk+sjr4SvvwlU0exn5fev8w9vDz+5/wCR1NFct/wlOq/9CnqOf95aP+Em1o/d8JXv4zKKPYz/AKa/zD28PP7n/kdTRXLf8JD4kb/V+EJT/v30a/0o/tfxc/3fC8Mf+/fof5Uexl3X3oPbx7P7n/kdTRXLfbPGsn3dL0qL/rpOzfyo8vxzN96fQ4B/sLIx/Wj2XeS+8Pbdov7jqaK5b+xvFc/+v8URwjusFkp/U80f8IbNP/x/eI9Yn9VSby1P4AUezgt5L8Q9pN7Qf3r/AIJ0k91b2qb7ieKFfWRwo/Wsa68aeHLPPmarA59Icyf+gg1DB4D8OwvvexNxJ3eeVnJ/AnH6Vs2ul6fY4+yWNtBj/nlEq/yFH7pd39y/zD98+y+9/wCRg/8ACai540zQtVvPR/I2If8AgRo+3eMr3/UaRYWCn+K6uDIR+CV1NFHtIraK+d3/AJB7Ob+KT+Vl/mcr/wAI/wCIr3nUPE8kanrHZQiPH0brUkPgPRFkEt2lxfzD/lpdzs5/LgfpXTUUe2n0dvTQPYU+qv66le0sLOwj2WlrDbp6RRhR+lWKKKybb1ZqklogooopDCiiigAooooAKKKKACiiigAooooAKKKKACiiigAooooAKKKKACiiigAooooAKKKKACiiigAooooAKKKKACiiigAooooAKKKKACiiigAooooAKKKKACsTXPDdvrBS5jka01KHmG7i4ZfY+o9q26KqMnF3RMoqStI5bT/E1xY3aaV4ljW1uzxFdD/U3HuD2Pt/LpXU1V1DTrTVLN7W9gSaF+qsOnuD2PvXMCPWvB/+qEuraIv8HW4tx7f3lH+cVraNT4dH26fL/Ix5pU/i1XfqvX/P7zsaKo6VrFhrVqLmwuEmT+IDhlPoR1Bq9WLTTszdNNXQUUUUhhRRRQAUUUUAFFFFABRRRQAUUUUAFFFFABRRRQAUUUUAFFFFABRRRQAUUUjMqKWYgKOSScAUALRWFfeMdA09tkmoxSSdBHBmRifT5c/rVL/hKdUvuNI8NXsinpLdkQL9eeorVUZtXtb10/Mxdamna935a/kdVSEhQSSAB1JrlvsPjHUP+PnVbLTUP8FpD5jY+rd/pSjwLY3BDarf6jqTdSLi4IT8AMY/On7OC+KX3a/5C9pN/DH79P8ANmleeKNCsMi41W1Vh1VX3sPwXJrN/wCE5tbnjS9M1PUCejQ25CfiT0/Ktaz8O6Np+Da6ZaxsOjeWC35nmtOi9JbJv5/5Baq92l6K/wCZy39peML3/j20OzsVPRry43/onNH9i+Kbvm88SJbqesdnbD/0I811NFHtbfCkvl/ncPY3+KTfz/ysct/wg1pNzf6rq16T1E10dv5CrEHgbw1APl0qJj6yMz/zNdDRSdao/tDVCkvsr+vUzI/DmiRD5NIsB7/Z0z/KitOio55dy+SPZfcFFct/wi+rz/8AH54s1Bs9fs6LD/LNH/CBaZL/AMfl5qd76/aLtjn8sVpyU1vL8H/wCOeo9o/e1/wTeuNU0+0z9pv7WHHXzJlX+ZrLn8beG7fO/VoDj/nmGf8A9BBpbfwV4ctseXpFu2P+emZP/Qia1INNsLXH2eytocf884lX+Qo/dLu/uX+Yfvn2X3v/ACMD/hPtHf8A49or+69PJtWOfzxR/wAJjNL/AMe3hnW39C9vsB/HNdTRRz0+kfxDkqdZfgct/wAJD4jl/wCPfwjN9ZrxE/Qij7d40l+5pGmwf9drgt/6DXU0Ue0j0ivx/wAw9lLrN/h/kct5Hjib793otv8A9co5Gx+dH9ieKpv9d4qWMf3YbFP5k5rqaKPbPol9yD2K6t/e/wDgHLf8Ijey/wDHz4p1hvXyZBH/ACo/4QLTH/4+bvU7n1867Y5/LFdTRR7ep0f5f5B7Cn1X5/5nNxeAvDMPI0tGPq8jt/M1fh8M6FB/q9HsQR3MCk/mRWrRUurUe8n95So01tFfcV4rGzh/1VrBH/uRgVYooqG29y0ktgooopDCiiigAooooAKKKKACiiigAooooAKKKKACiiigAooooAKKKKACiiigAooooAKKKKACiiigAooooAKKKKACiiigAooooAKKKKACiiigAooooAKKKKACiiigAooooAKKKKACiiigAooooAKKKKACiiigAooooAKKKKACiiigAooooAKKKKACiiigAooooA53VPCkNxdHUdLnbTdTHPnQj5ZPZ16H/PWq9t4puNMuEsfE9sLOVjtjvI+YJfx/hP1/SuqqK5tYLy3e3uYUmhcYZHXINbKpdcs1dfiYulZ81N2f4P8AruiRHV0DowZWGQQcgilrkW0DVfDzmbw3c+ba5y2m3LZX/gDH7p+v61f0nxZY6jP9juFew1EcNaXI2tn/AGT0b/PFDpac0NV/W6CNXXlmrP8AD5P+mb9FFFYmwUUUUAFFFFABRRRQAUUUUAFFFFABRRRQAUUVn6hrulaUD9uv4ICP4Wf5v++RyaaTbshOSirtmhRXK/8ACYzX/wAug6Je3+ek0g8mL/vpv/rUn9m+LNV5vtWg02E9YrFNz493PQ/StfYtfG0v67GPtk/gTf8AXdnRXmoWenRebe3UNunrK4XP0z1rn38b2ty5i0XT73VZAcboYysYPux6flU9n4J0S1l8+a3e+uO8145lY/geP0roERI0CIqqoGAqjAFF6Ue7/D/gharLsvxf+Ry+3xnqf3nsNHiP90efKP8A2WlXwNaXLB9Yv7/VH67Z5iqD6KOn511NFHtpL4dPT+rh7CL+LX1/y2KVjo+m6YuLKxt4PeOMAn6nqau0UVk227s2SSVkFFFFIYUUUUAFFFFABRRRQAUUUUAFFFFABRRRQAUUUUAFFFFABRRRQAUUUUAFFFFABRRRQAUUUUAFFFFABRRRQAUUUUAFFFFABRRRQAUUUUAFFFFABRRRQAUUUUAFFFFABRRRQAUUUUAFFFFABRRRQAUUUUAFFFFABRRRQAUUUUAFFFFABRRRQAUUUUAFFFFABRRRQAUUUUAFFFFABRRRQAUUUUAFFFFABRRRQAUUUUAFFFFABRRRQAUUUUAFFFFABRRRQAUUUUAFZ+q6Jp2tweTf2qTAfdY8Mv0YcitCimm07oTipKzOR+weJPDvOnXA1iwX/l2um2zIPRX7/j+VaGl+LdM1Kf7LIz2V8DhrW6XY+fbPB/Dmt6qGqaJputQeVqFpHMB91iMMv0YcitvaRl8a+a/qz/Ax9nKH8N/J/wCe6/Ev0VyX9k+IvD/zaPe/2lZr/wAud63zqPRJP8eKuad4w0+7uPsd6smm344NvdjZk/7LdD/Wk6TteOq/roNVle09H/XXY6GiiisTYKKKRmVFLOwVRySTgCgBaKwb7xnoFi3ltqCTS9BHbgysT6fLxVP/AISPXNQ40nw3OqHpNfuIgPfb1P4GtVRm9Wreun5mLrQTsnd+Wv5HVVDc3dtZxebdXEUEf96Vwo/M1zf9jeKNR51HX0s0PWLT4sf+PtyKmtvAuhxSedcwy38/eW8lMhP1HT9KfJBfFL7v6Queo/hj9/8AwLiT+OtFSUw2bXGoz/8APOzhMh/PgfrUX9reK9R4sNDgsYz0lv5sn/vheRXSwW0FrEIreGOGMdFjUKPyFS0c8F8Mfv8A6Qck38Uvu/ps5X/hGNX1DnWPEdyyHrBZKIV+mepH1rQ0/wAJaFphDW+nQmQc+ZKPMbPrls4/Ctqik602rXsvLT8ilRgne1356/mFFFFZGoUUUUAFFFFABRRRQAUUUUAFFFFABRRRQAUUUUAFFFFABRRRQAUUUUAFFFFABRRRQAUUUUAFFFFABRRRQAUUUUAFFFFABRRRQAUUUUAFFFFABRRRQAUUUUAFFFFABRRRQAUUUUAFFFFABRRRQAUUUUAFFFFABRRRQAUUUUAFFFFABRRRQAUUUUAFFFFABRRRQAUUUUAFFFFABRRRQAUUUUAFFFFABRRRQAUUUUAFFFFABRRRQAUUUUAFFFFABRRRQAUUUUAFFFFABRRRQAUUUUAFFFFABRRRQAUUUUAFU9R0qw1a38i/tY54+28cj6HqPwoopptO6E0mrM828SXl/wCBr+K30fULk2zDPkXDCVF9hkZA/GqH/C09f248qxz/AHvKbP8A6FRRXvUKUKlNSmrs+exFapSqyjBtIrH4geJL2dIvtywI5wfJiUH8yCa7zT/CFhqlrFd6rdahqDMM7Lm5JQfQDFFFc+MSpJez09DpwLdZv2mvqdLY6Vp+mrtsrKC3HrHGAT9T1NXKKK8ltt3Z7CSSsgooopDCiiigAooooAKKKKACiiigAooooAKKKKACiiigAooooAKKKKACiiigAooooA//2Q==","start":0,"end":249}]

**Comment by miguelgrinberg** [comment]
Are you sure your requests aren't passing through some intermediary that also has a maximum size? Like maybe nginx or some other proxy server?

**Reply by chenhuan-gao** [reply]
I do this locally, and I am new to flask socket io, so basically I just follow the tutorial

**Reply by chenhuan-gao** [reply]
I notice that, when it try to receive the 8mb file. It will get the packet (in log shows, Received packet MESSAGE data 51), and the disconnected immediately

**Reply by miguelgrinberg** [reply]
What client are you using?

**Reply by chenhuan-gao** [reply]
Python, I use flask socket io to build the communication between a python project and web pages

**Reply by miguelgrinberg** [reply]
Okay, great. Enable logging in the client and run the failing case again. Maybe there are some more clues in the client log.

**Reply by chenhuan-gao** [reply]
![image](https://github.com/miguelgrinberg/Flask-SocketIO/assets/38510078/1e583fd5-5292-4305-b930-f341a1426198)

I set like this, is there any other logger can be turned on?

**Reply by miguelgrinberg** [reply]
I'm asking about the client now, not the server.

**Reply by chenhuan-gao** [reply]
oh sry, the client is a web page use html, js, css. 

**here is the log:**
socket.io-client:manager writing packet {"type":2,"data":["upload learner recording",{"file":{},"name":"8m_test"}],"options":{"compress":true},"nsp":"/"} +10s
common.js:111 socket.io-parser encoding packet {"type":2,"data":["upload learner recording",{"file":{},"name":"8m_test"}],"options":{"compress":true},"nsp":"/"} +10s
common.js:111 socket.io-parser encoded {"type":5,"data":["upload learner recording",{"file":{"_placeholder":true,"num":0},"name":"8m_test"}],"options":{"compress":true},"nsp":"/","attachments":1} as 51-["upload learner recording",{"file":{"_placeholder":true,"num":0},"name":"8m_test"}] +0ms
common.js:111 engine.io-client:socket flushing 1 packets in socket +10s
common.js:111 engine.io-client:socket flushing 1 packets in socket +1ms
common.js:111 engine.io-client:socket socket close with reason: "transport close" +2s
common.js:111 socket.io-client:manager onclose +2s
common.js:111 socket.io-client:manager cleanup +0ms
common.js:111 socket.io-client:socket close (transport close) +12s
common.js:111 socket.io-client:manager will wait 898ms before reconnect attempt +0ms
common.js:111 socket.io-client:manager attempting reconnect +900ms
common.js:111 socket.io-client:manager readyState closed +0ms
common.js:111 socket.io-client:manager opening http://127.0.0.1:5000 +0ms
common.js:111 engine.io-client:socket creating transport "polling" +900ms
common.js:111 engine.io-client:socket options: {"path":"/socket.io/","agent":false,"withCredentials":false,"upgrade":true,"jsonp":true,"timestampParam":"t","rememberUpgrade":false,"rejectUnauthorized":true,"perMessageDeflate":{"threshold":1024},"transportOptions":{},"hostname":"127.0.0.1","secure":false,"port":"5000","query":{"EIO":4,"transport":"polling"},"socket":{"secure":false,"hostname":"127.0.0.1","port":"5000","transports":["polling","websocket"],"readyState":"opening","writeBuffer":[],"prevBufferLen":0,"opts":{"path":"/socket.io/","agent":false,"withCredentials":false,"upgrade":true,"jsonp":true,"timestampParam":"t","rememberUpgrade":false,"rejectUnauthorized":true,"perMessageDeflate":{"threshold":1024},"transportOptions":{},"hostname":"127.0.0.1","secure":false,"port":"5000"},"id":null,"upgrades":null,"pingInterval":null,"pingTimeout":null,"pingTimeoutTimer":null}} +0ms
common.js:111 engine.io-client:polling polling +13s

**Reply by chenhuan-gao** [reply]
![image](https://github.com/miguelgrinberg/Flask-SocketIO/assets/38510078/883847f8-9007-42d8-bb27-9fa707d77b67)

**Reply by miguelgrinberg** [reply]
Please create a small client and server example that I can use to reproduce the problem here, and I should be able to determine if there is an issue on my side or not.

**Reply by chenhuan-gao** [reply]
here is a toy example of the problem I facing:
https://github.com/chenhuan-gao/My_Toy_Example

When I try to upload the 219kb file and 2.58mb file, every thing works well, and with  13.9mb file, problem occured, "in upload file" won't be print and client will immediately disconnected.

**Reply by chenhuan-gao** [reply]
Flask                         1.1.2
Flask-SocketIO         5.3.4
Python                      3.9

**Reply by miguelgrinberg** [reply]
When using threading mode, with or without websocket everything works.
When using eventlet, it appears the eventlet's websocket implementation has a maximum of 8MB, which is reported clearly as an error to the client:

![image](https://github.com/miguelgrinberg/Flask-SocketIO/assets/2715854/4e288f3e-e87a-413b-8447-a2cccf5c9e69)

Are you using eventlet?

**Reply by chenhuan-gao** [reply]
yes！

**Reply by miguelgrinberg** [reply, marked as answer]
Please upgrade your `python-engineio` package for the fix for this issue.

**Reply by chenhuan-gao** [reply]
Thx a lot for your in time reply, and the problem has been solved!

**Comment by chenhuan-gao** [comment]
do you mean this?

**here is the log when I try to recieve the 8mb file:**
Bv_mju8y7_szCRN9AAAA: Received packet MESSAGE data 51-["upload learner recording",{"file":{"_placeholder":true,"num":0},"name":"8m_test"}]
client disconnected.
127.0.0.1 - - [04/Jul/2023 18:24:31] "GET /socket.io/?EIO=4&transport=websocket&sid=Bv_mju8y7_szCRN9AAAA HTTP/1.1" 200 0 224.093795
client connected.
xeFLV6HYnx9xhgpRAAAC: Sending packet OPEN data {'sid': 'xeFLV6HYnx9xhgpRAAAC', 'upgrades': ['websocket'], 'pingTimeout': 20000, 'pingInterval': 25000}
127.0.0.1 - - [04/Jul/2023 18:24:33] "GET /socket.io/?EIO=4&transport=polling&t=OaWRhDI HTTP/1.1" 200 278 0.001002
xeFLV6HYnx9xhgpRAAAC: Received packet MESSAGE data 0
xeFLV6HYnx9xhgpRAAAC: Sending packet MESSAGE data 0{"sid":"xBWiQxA3uwSbDduWAAAD"}

**Reply by miguelgrinberg** [reply]
This is a server log. I would like to see the client log.

**Comment by chenhuan-gao** [comment]
As you can see, the file has be received, but client disconnected immediately, so there is no error message. Do you have any idea what will triggler the disconection?

**Comment by bluthen** [comment]
Just a guess, maybe need to increase ping_timeout?

**Reply by chenhuan-gao** [reply]
just tried, and the problem is still the same.

**Reply by chenhuan-gao** [reply]
I set ping_timeout to 1000
